### PR TITLE
mfa: send username instead of SSH login name in MFA cert request

### PR DESCRIPTION
**Pull request by awly**
Username is the teleport username (either from SSO or for local user).
SSH login name is one of the OS logins allowed for the user.

In a user cert request, Username means the former, not the latter.

### CODE REVIEW

File: lib/client/client.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func (proxy *ProxyClient) IssueUserCertsWithMFA(ctx context.Context, params Reis
 	defer stream.CloseSend()
 
 	initReq := &proto.UserCertsRequest{
-		Username:          proxy.hostLogin,
+		Username:          cert.KeyId,
```
Comment (r0mant) [inline lib/client/client.go:378]: Not related to your change, just noticing that `RouteToApp` is not set in this request for some reason.

**Review by r0mant — APPROVED** [review]

File: lib/client/client.go (hunk 1 above)
Comment (awly) [reply lib/client/client.go:378]: https://github.com/gravitational/teleport/pull/5938 will end up setting it

**Review by russjones — APPROVED** [review]
Bot.

**Comment by awly** [comment]
https://github.com/gravitational/teleport/pull/5938 fixes this too, so I won't backport this PR and wait for that other one to be backported.

**Pull request merged**